src/backend/joanie/core/api/admin.py

```diff
@@ -40,9 +40,15 @@ class ProductViewSet(viewsets.ModelViewSet):
     authentication_classes = [SessionAuthenticationWithAuthenticateHeader]
     permission_classes = [permissions.IsAdminUser & permissions.DjangoModelPermissions]
     serializer_class = serializers.AdminProductSerializer
+    serializer_action_classes = {"list": serializers.AdminProductLightSerializer}
     queryset = models.Product.objects.all()
     filterset_class = filters.ProductAdminFilterSet
 
+    def get_serializer_class(self):
+        if self.action in self.serializer_action_classes:
+            return self.serializer_action_classes[self.action]
+        return self.serializer_class
```

Comment:
What do you think if we replace `serializer_action_classes` by `action_serializers` such as : 
```
action_serializers = {
        “list”: serializers.AdminProductLightSerializer
    }
```
and the method of `def get_serializer_class(self)` to :
```
def get_serializer_class(self):
        if hasattr(self, “action_serializers”):
            if self.action in self.action_serializers:
                return self.action_serializers[self.action]
        return super().get_serializer_class()
```

Reply:
If the `get_serializer_class` function was a class from a parent from class, I'd agree, but here I find that it adds complexity, don't you think?